### DISCUSSION TITLE
Listing monitors with just EventLoop instead of ActiveEventLoop

### DISCUSSION THREAD

**Discussion by bschwind** [rust-windowing/winit#3962] [Q&A] [answered]
This used to be possible quite awhile ago, but

* It was shifted from `EventLoop` to `EventLoopWindowTarget` in #1616
* In 0.30, `EventLoopWindowTarget` was renamed to `ActiveEventLoop`, and the `Deref` implementation from `EventLoop` to `EventLoopWindowTarget` was removed

The end result is that you can no longer list monitors with a plain `EventLoop`. Not the end of the world, but I happen to have an application which hasn't migrated to `ApplicationHandler` yet, and wants to query monitors before constructing a Window.

Is it still something that could be added back to `EventLoop`, or does it have soundness issues on some platforms and that's why it was removed?

**Comment by kchibisov** [comment]
You still can list monitors before creating window if you create window inside the event loop as suggested(you don't need to switch to the trait for that), you don't need application handler for that as well.

Listing monitors generally involves communication an sync thus it should have the loop _running_, this might not be the case on some platforms, but for cross platform API it was a bit unreliable.

**Reply by bschwind** [reply]
So my application also has its own loop and uses the [pump_events](https://docs.rs/winit/latest/winit/event_loop/struct.EventLoop.html#method.pump_events) function. I suppose I would still just listen for a `StartCause::Init` or `resumed` event in the closure, list the monitors there, then construct a window and store it on an `Option`al field?

> Listing monitors generally involves communication an sync thus it should have the loop running, this might not be the case on some platforms, but for cross platform API it was a bit unreliable.

Thanks for the explanation, that makes a lot of sense to move it to `ActiveEventLoop` if it can cause problems on some platforms.

**Reply by kchibisov** [reply, marked as answer]
yeah, you can use pump_events to poll the monitors and collect to e.g. Vector during `NewEvents`, and then do whatever you want.

**Reply by bschwind** [reply]
Makes sense, thanks for your help!
